release/v1.14 fails on Windows due to undeclared SO_DOMAIN

`release/v1.14` started failing on Windows on 6/7 (without any code changes), most likely due to changes in the AZP's Windows CI image (reorganized `include`, etc.):
```
source/common/network/utility.cc(354): error C2065: 'SO_DOMAIN': undeclared identifier
```
It still passes on `master`, and there are no changes between `release/v1.14` and `master` in `include/envoy/common/platform.h`, so I'm not sure where else to look, or how to narrow this down without access to Windows machine.

@lizan @mattklein123 any ideas on what might be the cause or who could help fix it?

--- a/.azure-pipelines/pipelines.yml
+++ b/.azure-pipelines/pipelines.yml
@@ -122,23 +122,3 @@ jobs:
           testResultsFiles: "**/bazel-testlogs/**/test.xml"
           testRunTitle: "macOS"
         condition: always()
-
-  - job: Windows
-    dependsOn: ["format"]
-    timeoutInMinutes: 360
-    pool:
-      vmImage: "windows-latest"
-    steps:
-      - powershell: |
-          .\ci\windows_ci_setup.ps1
-          Write-Host "##vso[task.prependpath]$env:TOOLS_BIN_DIR"
-        displayName: "Install dependencies"
-        env:
-          TOOLS_BIN_DIR: $(Pipeline.Workspace)\bin
-
-      - bash: ci/windows_ci_steps.sh
-        displayName: "Run Windows CI"
-        env:
-          TMPDIR: $(Agent.TempDirectory)
-          BAZEL_VC: "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\VC"
-          BAZEL_SH: "C:\\Program Files\\Git\\bin\\bash.exe"

